src/lib/alloc.c

```diff
@@ -429,12 +438,40 @@ static void *rmalloc_runtime(int zone, uint32_t caps, size_t bytes)
 		ptr = cache_to_uncache(ptr);
 
 	return ptr;
+}
+
+static void *rmalloc_runtime(int zone, uint32_t caps, size_t bytes)
+{
+	struct mm_heap *heap;
+	struct mm_heap *current = NULL;
+	void *ptr = NULL;
+
+	/* check runtime heap for capabilities */
+	do {
+		heap = get_runtime_heap_from_caps(caps, current);
+		if (heap) {
+			ptr = rmalloc_runtime_get(heap, zone, caps, bytes);
+			if (ptr)
+				return ptr;
+
```

Comment:
in rmalloc_runtime_get(), if alloc_block() return NULL. But if RZONE_FLAG_UNCACHED  is assigned, ptr = cache_to_uncache(ptr) will be called and it will be NULL now.

Reply:
good catch